src/world/agent.py

```diff
@@ -1,20 +1,24 @@
+import random
 from typing import Tuple
 
 import numpy as np
 
 from utils import Direction
 
+from world.virus import Virus
+
 
 class Agent:
-    def __init__(self, pos: Tuple, map, velocity: int = 4):
+    def __init__(self, pos: Tuple, region, virus: Virus = None, velocity: int = 4):
         self.pos = pos
         self.age = 0
-        self.is_sick = False
-        self.sick_time = 0
-        self.remaining_immunity = 0
+        self.is_sick = virus is not None
+        self.sick_time = virus.sick_time if virus else 0
+        self.virus = virus
+        self.remaining_immunity = [virus.immunity_time, virus] if virus else [None, None]  # (immunity, last_virus_name)
```

Comment:
I think `remaining_immunity` should be a list of tuples so that the agent is immune to more than one virus. WDYT?  
I'll add this in next PR.